vllm/executor/ray_gpu_executor.py

```diff
@@ -40,11 +55,7 @@ def _init_executor(self) -> None:
         # Create the parallel GPU workers.
         self._init_workers_ray(placement_group)
 
-        self.forward_dag = None
-        if USE_RAY_COMPILED_DAG:
-            self.forward_dag = self._compiled_ray_dag()
-            self.extra_execute_model_run_workers_kwargs[
-                "use_ray_compiled_dag"] = True
+        self.forward_dag: Optional["ray.dag.CompiledDAG"] = None
```

Comment:
nit: you can remove None if we know it is lazily initialzed and won't be used before that.

```
self.forward_dag: "ray.dag.CompiledDAG"
```

This can remove Optional type which is much cleaner. example: https://github.com/vllm-project/vllm/blob/fc17110bbef4e78703abffac51133a2fb71e9f79/vllm/worker/model_runner.py#L244

Reply:
Thanks for the suggestion. I think the reason we need to set it to `None` here is that later we have a `None` check. With type annotation only the value is not set to `None`.